### ISSUE TITLE
Route instance no longer accessible to getComponents in 2.2.0

### ISSUE THREAD

**Issue by kpdecker** [reactjs/react-router#3305]
## Version
2.2.0

## Test Case
```
    <Route
        path={path}

        getComponent={function(location, callback) {
          loader((routes) => {
            this.scriptPath = routes.path.replace(/^.*\//, '');
            callback(undefined, routes.rootComponent);
          });
        }}
        getIndexRoute={function(location, callback) {
            this.scriptPath = routes.path.replace(/^.*\//, '');
            callback(undefined, routes.indexRoute);
        }}
    />
```

## Steps to reproduce
Attempt to access route object through `this` in `getComponent` or `getComponents`.

## Expected Behavior
The current route is available on the `this` object as was implemented in previous versions and is also implemented for helpers like `getIndexRoute`. Alternatively, this is explicitly exposed somewhere.

## Actual Behavior
`this` is undefined.

**Comment by taion** [comment]
Oops, my bad.

**Comment by taion** [comment]
I have a fix in #3306. I'll release it as soon as another collaborator signs off on the change. Sorry for the regression. The PR adds a test case to catch this in the future.

**Comment by kpdecker** [comment]
Thanks for the prompt fix!

**Comment by taion** [comment]
Released as v2.2.1.
